TopologyId.asciidoc

```diff
@@ -0,0 +1,40 @@
+== Topology Id Rework
+
+The Infinispan Hot Rod client is able to know which node to direct requests to based on the key.
+This has to be updated every time the topology changes (new node leaves or joins the cluster).
+When the topology changes a counter is incremented that the server is aware of to signal this.
+This client then is able to tell the server which topology id it last knew about and in the case it is
+old the server will reply with the new id and the members.
+
+Unfortunately, this has an interesting problem if the entire server cluster is shut down but a client
+is still around.
+The client will of course not function when the cluster is gone, however when the server starts back up
+the server topology id will be reset back to 0.
+This means a client will have a _higher_ topology so the server will never send the cluster information
+and the client will, until the server toplogy catches up, no longer know what servers a key map to and
+also send to servers that may not even exist any longer.
+
+=== Suggested Fix
+
+Fixing this will require a few different things.
+
+* Change the toopology id to be a more unique number that can be recreated instead of an incrementing number
+** Possible choices are to hash the server member ids
```

Comment:
better hashing the mappings (segments => owners) since the state transfer uses 2 topology changes.

Reply:
Yes, sorry I didn't mean this as the answer but just a suggestion. I should have tried to go into more specifics. Let me update with some actual options.